cdisc_rules_engine/rules_engine.py

```diff
@@ -134,9 +138,21 @@ def validate_single_rule(
                 is_split_dataset(datasets, dataset_domain),
                 datasets,
             ):
-                result: List[Union[dict, str]] = self.validate_rule(
-                    rule, dataset_path, datasets, dataset_domain
-                )
+                result = None
+                if self.is_validate_xml:
```

Comment:
do we need to check this here? If an exception occurs won't it be logged in the outer `except` block and logged? If the goal is to create an empty results list, you will need to set result to an empty list when catching the exception, but I think we can just check this flag in the outer exception block